test/ampycloud/test_data.py

```diff
@@ -228,6 +253,36 @@ def test_ceilochunk_nocld():
     assert chunk.metar_msg() == 'NCD'
 
 
+def test_ceilochunk_highcld():
+    """ Test the methods of CeiloChunks when high clouds are seen in the interval. """
+
+    dynamic.AMPYCLOUD_PRMS['MAX_HITS_OKTA0'] = 3
+    dynamic.AMPYCLOUD_PRMS['MSA'] = 10000
+
+    n_ceilos = 4
+    lookback_time = 1200
+    rate = 30
+    # Create some fake data to get started
+    # 1 very flat layer with no gaps
+    mock_data = mocker.mock_layers(
+        n_ceilos, lookback_time, rate,
+        [{'alt': 15000, 'alt_std': 10, 'sky_cov_frac': 0.1, 'period': 100, 'amplitude': 0}]
+    )
+
+    # Instantiate a CeiloChunk entity ...
+    chunk = CeiloChunk(mock_data)
+
+    # Do the dance ...
+    chunk.find_slices()
+    chunk.find_groups()
+    chunk.find_layers()
+
+    # Assert the final METAR code is correct
+    assert chunk.metar_msg() == 'NSC'
```

Comment:
Could be good to also add a test when there is a cloud layers within the MSA buffer, just to make sure we issue 'NSC' in this case as well.

Reply:
I added one here by parameterizing the mock layers, one within thr buffer and one above.